### PR TITLE
Move ErrorResponse into HttpPipelineExtensions

### PR DESCRIPTION
**Pull request by annelo-msft**
Follow up to https://github.com/Azure/azure-sdk-for-net/pull/38509

### CODE REVIEW

**Comment by annelo-msft** [comment]
Running Autorest Regen Preview: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3061775&view=results

**Comment by annelo-msft** [comment]
Merged change branch and re-running generated PR: https://github.com/Azure/azure-sdk-for-net/pull/38541

File: eng/DownloadSharedSource.ps1 [hunk 1]
```diff
@@ -19,8 +19,8 @@ Write-Host "git sparse-checkout init"
 git sparse-checkout init
 Write-Host "git sparse-checkout set --no-cone $azCoreSharedPath $armCoreSharedPath"
 git sparse-checkout set --no-cone $azCoreSharedPath $armCoreSharedPath
-Write-Host "git checkout"
-git checkout
+Write-Host "git checkout users/annelo/core-reshuffle-2"
```
Comment (m-nash) [inline eng/DownloadSharedSource.ps1:22]: update back to main before merging

**Review by m-nash — APPROVED** [review]

File: eng/DownloadSharedSource.ps1 (hunk 1 above)
Comment (annelo-msft) [reply eng/DownloadSharedSource.ps1]: Done - thanks!

**Pull request merged**